File: adminportal/src/Dto/AdminPortal/ThreadInfo.ts
import type { PostBasicInfo } from "./PostInfo"
import type { TopicBasicInfo } from "./TopicInfo"
import type { UserBasicInfo } from "./UserInfo"

export type ThreadEntry = {
    threadId: string,
    threadName: string,
    createdDate: Date,
    topicId: string,
    createdByUserId: string,
    numberOfPosts: number,
    hasNewPosts: boolean,
}

export type ThreadBasicInfo = {
    thread: ThreadEntry,
    totalPosts: number,
    newestMessage: PostBasicInfo,
    createdByUser: UserBasicInfo,
    topic?: TopicBasicInfo
}

export type ThreadSummary = {
    totalThreads: number
}

File: server/RestApiServer.Endpoints/Dto/Forum/ThreadInfo.cs
using RestApiServer.Db;
using RestApiServer.Dto.App;

namespace RestApiServer.Dto.Forum
{
    public class ThreadBasicInfo
    {
        public required ThreadEntry Thread { get; set; } = null!;
        public int TotalPosts { get; set; } = 0;
        public PostBasicInfo NewestMessage { get; set; } = null!;
        public UserBasicInfo CreatedByUser { get; set; } = null!;
        public TopicBasicInfo? Topic { get; set; } 
    }

    public class ThreadFullInfo 
    {
        public required ThreadEntry Thread { get; set; } = null!;
        public required int TotalPosts { get; set; } = 0;
        public required List<PostBasicInfo> Posts { get; set; }
        //Represents the user that created the thread
        public required UserBasicInfo CreatedByUser { get; set; }
    }

    public class ThreadSummary
    {
        public required int TotalThreads { get; set; }
    }
}